src/states/node.rs

```diff
@@ -1654,6 +1736,64 @@ impl Node {
         true
     }
 
+    fn send_own_group_merge(&mut self,
+                            targets: Vec<XorName>,
+                            merge_details: OwnMergeDetails<XorName>,
+                            src: Authority) {
+        for target in &targets {
+            let (connected, needed): (Vec<_>, Vec<_>) = merge_details.groups
+                .clone()
+                .into_iter()
+                .partition(|&(prefix, _)| merge_details.prefix.is_compatible(&prefix));
```

Comment:
I agree with your first point - good catch.  I had been thinking all groups falling under the merged prefix would be fully interconnected, and was trying to save some bandwidth.  I think this would be a valid optimisation if instead of `merge_details.prefix.is_compatible(&prefix)` we did `<target's prefix>.is_compatible(&prefix)`, but I'll just take the approach you suggest and pass the full list of `PublicId`s to each target.

I'm not sure about the second point.  Won't the group triggering the merge always be connected to all the neighbouring groups falling under the new merged prefix?  Any groups outside the new prefix will be handled by the `OtherGroupMerge`s.